test/linalg/matmul.jl

```diff
@@ -5,189 +5,279 @@ using Base.Test
 ## Test Julia fallbacks to BLAS routines
 
 # matrices with zero dimensions
-for i = 1:10
```

Comment:
I think it fine to remove the loop. We should see occational test failures even without it if the arrays are not set correctly.